provider/maas/environ.go

```diff
@@ -2134,17 +2141,24 @@ func (env *maasEnviron) allocateContainerAddresses1(hostInstanceID instance.Id,
 		deviceNICIDs[i] = maasNICID
 		subnetID := string(nic.ProviderSubnetId)
 
+		if !knownSubnet {
+			continue
+		}
+
 		linkedInterface, err := env.linkDeviceInterfaceToSubnet(deviceID, maasNICID, subnetID, modeStatic)
 		if err != nil {
-			return nil, errors.Annotate(err, "cannot link device interface to subnet")
+			logger.Warningf("linking NIC %v to subnet %v failed: %v", nic.InterfaceName, nic.CIDR, err)
```

Comment:
If we can't link the device to the subnet is it really just a warning, or is it more severe?